tests/unit/cluster/replica_migration.tcl

```diff
@@ -0,0 +1,141 @@
+# Allocate slot 0 to the last primary and evenly distribute the remaining
+# slots to the remaining primary.
+proc my_slot_allocation {masters replicas} {
+    set avg [expr double(16384) / [expr $masters-1]]
+    set slot_start 1
+    for {set j 0} {$j < $masters-1} {incr j} {
+        set slot_end [expr int(ceil(($j + 1) * $avg) - 1)]
+        R $j cluster addslotsrange $slot_start $slot_end
+        set slot_start [expr $slot_end + 1]
+    }
+    R [expr $masters-1] cluster addslots 0
+}
+
+start_cluster 4 4 {tags {external:skip cluster} overrides {cluster-node-timeout 1000 cluster-migration-barrier 999}} {
+   test "Write some data to primary 0, slot 1, make a small repl_offset" {
+        for {set i 0} {$i < 1024} {incr i} {
+            R 0 incr key_991803
+        }
+        assert_equal {1024} [R 0 get key_991803]
+    }
+
+    test "Write some data to primary 4, slot 0, make a big repl_offset" {
+        for {set i 0} {$i < 10240} {incr i} {
+            R 3 incr key_977613
+        }
+        assert_equal {10240} [R 3 get key_977613]
+    }
+
+    test "If a replica has not completed sync, it can not do the failover" {
+        # 10s, make sure primary 1 will hang in the save
+        R 0 config set rdb-key-save-delay 100000000
```

Comment:
the nosave does't matter. i am removing it. this rdb-key-save-delay thing is aim to slow down the RDB fork save.

Reply:
I see. So is the idea here to slow down the full sync but not about the shutdown? if so, maybe keep "nosave" on the shutdown path to speed up the test?

Reply:
yes, the idea is to slow down the full sync. Keeping nosave or not keeping both ok, currently the test code did not enable the "save", so the shutdown won't trigger the save. i can re-add it since it is no harm.